src/main/java/Mona/Storage.java

```diff
@@ -0,0 +1,96 @@
+package mona;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * This class contains the logic for interacting with the data logs
+ */
+public class Storage {
+    protected static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("M/d/yyyy HHmm");
+    private String filePath;
+
+    /**
+     * Constructor for Storage
+     * @param filePath location for the logs to be written to
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        String workingDirectory = System.getProperty("user.dir");
+        try {
+            Path dataDirectory = Paths.get(workingDirectory + "/data");
+            Files.createDirectories(dataDirectory);
+            File logFile = new File(workingDirectory + "/" + this.filePath);
+            logFile.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Error occurred setting up log" + e.getMessage());
+        }
+    }
+
+    /**
+     * Method for adding the tasks stored in the logs to Mona's task list
+     * @param currentTasks Mona's task list for tasks stored in the logs to be added to
+     */
+    public void readLog(List<Task> currentTasks) {
+        File log = new File(this.filePath);
+        try {
+            Scanner s = new Scanner(log);
+            while (s.hasNext()) {
+                String[] task = s.nextLine().split("\\|");
+                currentTasks.add(parseLogEntry(task));
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found:" + e.getMessage());
+        }
+    }
+
+    /**
+     * Method for translating the tasks stored in the logs into their corresponding Tasks
```

Comment:
Rather than returning null, would it be better to throw some form of ParserException here?